ChangeLog.d/fix-gcc12-warnings.txt

```diff
@@ -0,0 +1,2 @@
+Bugfix
+   * Fix GCC12 compiler warning about variable being used uninitialised.
```

Comment:
Playing devil's advocate: do we need a ChangeLog entry for a change to quiet a compiler warning in the tests?

Reply:
Well, I was not sure, but the IAR changes did not get through without....